waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/ThriftMetastoreClient.java

```diff
@@ -217,15 +218,23 @@ public void close() {
     }
     // Transport would have got closed via client.shutdown(), so we don't need this, but
     // just in case, we make this call.
-    if (isOpen()) {
+    if (transport != null && transport.isOpen()) {
       transport.close();
       transport = null;
     }
     LOG.info("Closed a connection to metastore, current connections: " + CONN_COUNT.decrementAndGet());
   }
 
   public boolean isOpen() {
-    return transport != null && transport.isOpen();
+    if (transport != null && transport.isOpen() && client != null) {
+      try {
+        client.getStatus();
+        return true;
+      } catch (TException e) {
+        return false;
```

Comment:
Do we want to log something if this happens?

Reply:
good question, I'll have a look if it isn't printed already, it might be due to how the logic is wired but perhaps it needs some cleaning up

Reply:
I changed the logic around a bit, but the error gets correctly logged in the logs (for every retry)